File: backend/src/scripts/seedCourses.ts
import fs from "node:fs";
import path from "node:path";
import { getDb, getCourseCountsByCountry } from "../services/db";

const SEED_FILE = path.resolve(__dirname, "../../data/seed_courses.sql");

function main() {
  const db = getDb();

  const existing = db.prepare(`SELECT COUNT(*) as c FROM courses`).get() as { c: number };
  if (existing.c > 0) {
    console.log(`courses table already has ${existing.c} rows — clearing before reseeding.`);
    db.exec(`DELETE FROM courses;`);
  }

  const sql = fs.readFileSync(SEED_FILE, "utf8");
  db.exec(sql);

  const total = (db.prepare(`SELECT COUNT(*) as c FROM courses`).get() as { c: number }).c;
  console.log(`Seeded ${total} courses.`);

  console.log("By country:");
  for (const row of getCourseCountsByCountry(db)) {
    console.log(`  ${row.country}: ${row.count}`);
  }
}

main();

File: backend-py/app/scripts/seed_courses.py
from pathlib import Path

from app.db import get_course_counts_by_country, get_db

_SEED_FILE = Path(__file__).resolve().parent.parent.parent / "data" / "seed_courses.sql"


def main() -> None:
    conn = get_db()

    existing = conn.execute("SELECT COUNT(*) as c FROM courses").fetchone()["c"]
    if existing > 0:
        print(f"courses table already has {existing} rows — clearing before reseeding.")
        conn.execute("DELETE FROM courses")

    sql = _SEED_FILE.read_text()
    conn.executescript(sql)
    conn.commit()

    total = conn.execute("SELECT COUNT(*) as c FROM courses").fetchone()["c"]
    print(f"Seeded {total} courses.")

    print("By country:")
    for row in get_course_counts_by_country(conn):
        print(f"  {row['country']}: {row['count']}")


if __name__ == "__main__":
    main()
